### PR TITLE
Add Dockerfile

### LINKED ISSUE #8: Add Dockerfile
Add Dockerfile and Makefile targets to build cdf, ctf, hss, hssgtw, pcrf images.

Depends-on #9

### PR DESCRIPTION
**Pull request by krsna1729**
Added Dockerfile to build c3po NF images
Added Makefile targets to build and push images
Added install_deps.sh

Fixes: #8, #9
Based-on: #16

Signed-off-by: Saikrishna Edupuganti <saikrishna.edupuganti@intel.com>

### CODE REVIEW

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    rm -rf /var/lib/apt/lists/*
+COPY --from=build /c3po/pcrf/bin /bin
+
+## Stage pcef
+FROM $RUN_BASE as pcef
+RUN bash -c "source ./install_rundeps.sh && install_run_pcef_deps" && \
+    rm -rf /var/lib/apt/lists/*
+COPY --from=build /c3po/pcrf/pcef/bin /bin
+
+## Stage pcrf
```
Comment (stolsma) [inline Dockerfile]: Pcrf should be tdf?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+FROM $RUN_BASE as tdf
+RUN bash -c "source ./install_rundeps.sh && install_run_tdf_deps" && \
+    rm -rf /var/lib/apt/lists/*
+COPY --from=build /c3po/pcrf/tdf/bin /bin
+
+## Stage pcrf
```
Comment (stolsma) [inline Dockerfile]: Same as above but then tssf?

**Comment by krsna1729** [comment]
@stolsma Thanks for reviewing!! Updated :)

**Comment by krsna1729** [comment]
retest this please

**Review by somnathchakrabarti — APPROVED** [review]

**Review by somnathchakrabarti — APPROVED** [review]
OK to merge

**Comment by hyunsun** [comment]
Can we merge this patch?

**Comment by krsna1729** [comment]
@omec-project/c3po-maintainers

**Comment by suresh-marikkannu** [comment]
retest this please

**Comment by suresh-marikkannu** [comment]
retest this please

**Comment by krsna1729** [comment]
@suresh-marikkannu this is dependent on #16 and #27

**Comment by hyunsun** [comment]
@krsna1729 Here is the standard docker build method we use in CORD project. Please update Makefile accordingly so that docker images in dockerhub can be updated automatically whenever new patch is merged or new version is released. https://guide.opencord.org/developer/test_release_software.html#publish-docker-container-images-to-public-dockerhub-job-docker-publish

**Comment by hyunsun** [comment]
retest this please

**Comment by krsna1729** [comment]
@hyunsun i think this and makefile pull request are failing because the CI needs to be updated to use .ci scripts in this repo instead of doing somehting manual.

**Comment by krsna1729** [comment]
https://github.com/omec-project/omec-project-ci/pull/36

**Comment by hyunsun** [comment]
@krsna1729 Thanks. I really hope this PR gets merged as soon as possible.

**Review by suresh-marikkannu — APPROVED** [review]

**Pull request merged**